src/stream/src/task/stream_manager.rs

```diff
@@ -228,25 +228,26 @@ impl LocalStreamManager {
         result
     }
 
-    pub async fn sync_epoch(&self, epoch: u64) -> (Vec<LocalSstableInfo>, bool) {
+    pub async fn sync_epoch(&self, epoch: u64) -> Result<(Vec<LocalSstableInfo>, bool)> {
```

Comment:
How will upper layer handle this error? Will it trigger the recovery process?

Reply:
Meta will get the error via the `BarrierComplete` rpc response and will trigger the recovery process.

FYI.
```
    async fn barrier_complete_and_commit(
        &self,
        prev_epoch: u64,
        result: MetaResult<Vec<BarrierCompleteResponse>>,
        state: &mut BarrierManagerState,
        tracker: &mut CreateMviewProgressTracker,
        checkpoint_control: &mut CheckpointControl<S>,
    ) {
        if let Err(err) = result {
            fail_point!("inject_barrier_err_success");
            let fail_node = checkpoint_control.fail();
            tracing::warn!("Failed to commit epoch {}: {:?}", prev_epoch, err);
            self.do_recovery(err, fail_node.into_iter(), state, tracker)
                .await;
            return;
        }
        ...
      }
```